# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/kaant7/flyrank-internship-ml/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose Lane 1: Ranking Signal Analysis because the starter dataset (content_refresh_anonymized.csv) has both observed signals (impressions, clicks, position, engagement, freshness) and a trend label (trend_direction). This makes it a good starting point for discovering which signals actually associate with visibility, clicks, or movement — before jumping into scoring or clustering.

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

What decision does your work improve? It improves the decision of which pages the content team should prioritize for review — instead of scanning the full inventory manually, the analysis highlights which observed signals (impressions, position, freshness, CTR, engagement) are actually associated with visibility or performance movement, so reviewers know what to look for.

Who acts on it? A content editor or SEO reviewer, who uses the signal report to decide which pages go into their review queue.

What does a wrong recommendation cost? Two kinds of cost: false positives waste reviewer time on pages that aren't real problems, and false negatives mean a real decline gets missed because a meaningful signal was dismissed as noise — both slow down the team's ability to act where it actually matters.

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

# Apply the starter feature-prep rules from the lane guide
df = df[(df["impressions_90d"] > 0) & (df["content_age_days"] >= 90)]
df = df.drop_duplicates(subset="content_id")

print("Total usable rows:", len(df))

# Number 1: enough scale to find real patterns, not a toy sample
declining_pct = (df["trend_direction"] == "down").mean() * 100
print(f"Share of pages currently declining: {declining_pct:.1f}%")

# Number 2: meaningful variation — not everything is the same, so there's something to explain
high_imp = df[df["impressions_90d"] >= 500]
low_imp = df[df["impressions_90d"] < 500]
print("Decline rate, high-impression pages:", (high_imp["trend_direction"] == "down").mean() * 100)
print("Decline rate, low-impression pages:", (low_imp["trend_direction"] == "down").mean() * 100)

# Number 3: a real, non-zero relationship between two observed signals
print("CTR - position correlation:", df["ctr"].corr(df["avg_position"]))

Total usable rows: 30000
Share of pages currently declining: 54.2%
Decline rate, high-impression pages: 59.55398780341983
Decline rate, low-impression pages: 47.46873587464216
CTR - position correlation: -0.07259029976832472


Why this looks worth it: With 30,000 usable pages, there's enough scale to detect real patterns rather than noise. Decline rates differ meaningfully between high- and low-impression pages (59.6% vs 47.5%) — an 12-point gap suggests visibility level isn't independent of trend direction, exactly the kind of signal-to-outcome relationship Lane 1 is built to investigate. The CTR-position correlation (-0.073) is weak but in the expected direction, worth digging into further with position-tier-adjusted comparisons rather than a raw correlation.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What my work will be able to say:

-Observed: Which signals (impressions, position, freshness, CTR, engagement) show measurable association with visibility and trend movement in this dataset.

-Directional: For example, pages with higher impressions show a higher decline rate than lower-impression pages (59.6% vs 47.5%), suggesting visibility level and trend direction are not independent.

-Decision-support: A ranked or flagged set of signals that a reviewer can use to prioritize which pages to look at first — narrowing down where limited review time goes, not making the call for them.

What my work will never say:

-Causal proof: A correlation between a signal and decline does not mean that signal caused the decline — I did not run an experiment, so I cannot claim cause and effect.

-"Predicting Google": I am not claiming to have found or reverse-engineered a Google ranking factor. These are observed correlations within FlyRank's own data, not evidence about how any search engine's algorithm works.

-Guaranteed outcomes: A page flagged as high-risk or high-opportunity is a candidate for review, not a guarantee that fixing it will recover traffic.

In [7]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.